### DISCUSSION TITLE
TypeLoadException for Nuke.Docker when using DockerTasks

### DISCUSSION THREAD

**Discussion by pistotniks** [nuke-build/nuke#696] [Q&A] [answered]
I have the following code in my target CreateDatabase that looks like compileable in Visual Studio:

```
  Target CreateDatabase => _ => _
    .Executes(() =>
    {
      DockerTasks.DockerCommit(x => x
        .SetContainer($"database")
        .SetRepository("myazure.io/databases/mydb")
      );
    });
```

and in my csproj I have
    TargetFramework netcoreapp3.1 TargetFramework
    PackageReference Include="Nuke.Common" Version="5.0.2"
    PackageReference Include="Nuke.Docker" Version="0.4.0"

First reference came in by default and the second reference I have added with using nuget package manager to try to install Nuke.Docker library. So this is all valid by me.

When I run:

nuke CreateDatabase

I get first the following warnings:
PowerShell Desktop version 5.1.18362.1171
Microsoft (R) .NET Core SDK version 5.0.103
C:\git\myproject\build\_build.csproj : warning NU1608: Detected package version outside of dependency constraint: Nuke.Docker 0.4.0 requires Nuke.Common (>= 0.20.1 && < 1.0.0) but version Nuke.Common 5.0.2 was resolved.
C:\git\myproject\build\_build.csproj : warning NU1608: Detected package version outside of dependency constraint: Nuke.Docker 0.4.0 requires Nuke.Common (>= 0.20.1 && < 1.0.0) but version Nuke.Common 5.0.2 was resolved.

and then I get the following error in powershell shell:

TypeLoadException: Method 'get_ProcessCustomLogger' in type 'Nuke.Docker.DockerCommitSettings' from assembly '**Nuke.Docker, Version=1.0.0.0**, Culture=neutral, PublicKeyToken=null' does not have an implementation.
   at Build.<get_CreateDatabase>b__33_1()
   at Nuke.Common.Execution.BuildExecutor.<>c.<Execute>b__4_0(Action x)
   at Nuke.Common.Utilities.Collections.EnumerableExtensions.ForEach[T](IEnumerable`1 enumerable, Action`1 action)
   at Nuke.Common.Execution.BuildExecutor.Execute(NukeBuild build, ExecutableTarget target, IReadOnlyCollection`1 previouslyExecutedTargets, Boolean failureMode)

I know I should be able to solve this problem on my own if I know just a little bit of ABC of c# LOL, but this is so magical for the moment to me. Does somebody have an idea what should I actually do? Where is this Nuke.Docker, Version=1.0.0.0 coming from although I have 0.4.0 as a reference? The version 1.0.0 does not even exist, so how could I be even getting this message as a feedback?
https://github.com/arodus/nuke-docker/blob/master/CHANGELOG.md

**Comment by matkoch** [comment, marked as answer]
The `Nuke.Docker` package is deprecated. Where did you get that from?

**Reply by pistotniks** [reply]
Hm, interesting. I apologize, since I am new to this, I am still figuring out how it works. So I tried to use the DockerTasks abstraction in my Visual Studio and Resharper did not give me any option to import the class, so I immediately went to "Tools/Nuget package manager" and found Nuke.Docker lib to be installed. And this is what happened. It installed this deprecated version 0.4.0.. I see now that all I need is to remove the deprecated library and use the one from Nuke.Common.Tools.Docker. Thank you for your help.
